library/alloc/src/vec/mod.rs

```diff
@@ -1679,6 +1679,50 @@ impl<T, A: Allocator> Vec<T, A> {
         }
     }
 
+    /// Appends an element if there is sufficient spare capacity, otherwise the element is returned.
+    ///
+    /// Unlike [`push`] method will not reallocate when there's insufficient capacity.
+    /// The caller should use [`reserve`] or [`try_reserve`] to ensure that there is enough capacity.
+    ///
+    /// [`push`]: Vec::push
+    /// [`reserve`]: Vec::reserve
+    /// [`try_reserve`]: Vec::try_reserve
+    ///
+    /// # Examples
+    ///
+    /// A manual, panic-free alternative to FromIterator
+    ///
+    /// ```
+    /// #![feature(vec_push_within_capacity, try_reserve)]
+    ///
+    /// use std::collections::TryReserveError;
+    /// fn from_iter<T>(iter: impl Iterator<Item=T>) -> Result<Vec<T>, TryReserveError> {
+    ///     let mut vec = Vec::new();
+    ///     for value in iter {
+    ///         if let Err(value) = vec.push_within_capacity(value) {
+    ///             vec.try_reserve(1)?;
+    ///             // this cannot fail, the previous line either returned or added at least 1 free slot
```

Comment:
Use `.unwrap()` to show that this cannot fail instead of ignoring the error.

Reply:
I wrote it this way to avoid panic codegen which can be one of the benefits of this method.